clippy_utils/src/lib.rs

```diff
@@ -2234,7 +2234,7 @@ pub fn is_slice_of_primitives(cx: &LateContext<'_>, expr: &Expr<'_>) -> Option<S
 }
 
 /// returns list of all pairs (a, b) from `exprs` such that `eq(a, b)`
-/// `hash` must be comformed with `eq`
+/// implies `hash(a) == hash(b)`
```

Comment:
Should be 'Requires `eq` and `hash` be functions such that `eq(a, b)` implies `hash(a) == hash(b)`.' It's a precondition to the function, not a description of what the function does.

Reply:
I took a stab at it.

Reply:
Just flip the two paragraphs and then it's fine. Weird to have details of how to use the function before knowing what it does.

Sorry for the delay here. Been busy this week.

Reply:
I'm in no hurry and I appreciate that people are taking time from their normal work to review these changes.

I've squashed all the spelling changes into a single commit and left this change as its own, but if you'd like, I can squash it in as well.

Reply:
It works as a separate commit.